### PR TITLE
Getting started with KMM

### PR DESCRIPTION
**Pull request by mohit-sharma-87**
https://github.com/mohit-sharma-87/DevHub-Articles/blob/get_started_kmm/Getting%20Started%20with%20KMM.md

### CODE REVIEW

File: Getting Started with KMM.md [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+        setupRealmSync()
+    }
+    return realm.query<QueryInfo>().asFlow().map {
+        it.list.map { it.queries }
+    }.asCommonFlow()
+}
+```
+
+With this done our application is ready to send and receive data in real time. Yes in real time,
+no additional implementation is required to make it realtime.
```
Comment (DominicFrei) [inline Getting Started with KMM.md:548]:
```suggestion
With this done our application is ready to send and receive data in real-time. Yes, in real-time. No additional implementation is required to make it real-time.
```

To be absolutely precise, it's not real-time, but it's freakingly fast. :)

File: Getting Started with KMM.md [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+
+Also, you view or modify the data received via `saveInfo` function using `Atlas` UI.
```
Comment (DominicFrei) [inline Getting Started with KMM.md:550]:
```suggestion
Also, you can view or modify the data received via the `saveInfo` function using the `Atlas` UI.
```

File: Getting Started with KMM.md [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+![Atlas UI](https://mongodb-devhub-cms.s3.us-west-1.amazonaws.com/Screenshot_2023_01_10_at_09_17_27_2a1a38dece.png)
```
Comment (DominicFrei) [inline Getting Started with KMM.md:551]: I just noticed it here but it also targets all other screenshots.
I'd leave an empty line before the screenshot in your markdown. That way the text above it does not stick to the screenshot directly. Unless you want that. :)

**Review by DominicFrei — CHANGES_REQUESTED** [review]
Nice one, @mohit-sharma-87 ! 💪 

I didn't see anything major but left some comments. Hope they help. :)

File: Getting Started with KMM.md [hunk 4]
```diff
@@ -0,0 +1,564 @@
...
+
+<img src="https://mongodb-devhub-cms.s3.us-west-1.amazonaws.com/first_multiplatform_project_on_android_1_e53c9f0ee4.png"  width=40%> <img 
+src="https://mongodb-devhub-cms.s3.us-west-1.amazonaws.com/first_multiplatform_project_on_ios_1_cda32945d8.png"  width=40%> 
+
+## Basic of the Kotlin Multiplatform
+
+Now it's time to understand what is happening under the hood to grasp the basic concepts of KMM.
+
+Kotlin Multiplatform is all about code sharing within apps for different environments (iOS, Android,
+etc). Some common use cases for shared code are getting data from the network, saving it into the
```
[the remark this answers is not in the document]
Comment (mohit-sharma-87) [reply Getting Started with KMM.md:73]: removed etc.

File: Getting Started with KMM.md [hunk 5]
```diff
@@ -0,0 +1,564 @@
...
+device, filtering or manipulating data, etc. This is different from other
+cross-development frameworks as this enforces developers to share only business logic code rather
+than complete code which often makes things complicated.
```
[the remark this answers is not in the document]
Comment (mohit-sharma-87) [reply Getting Started with KMM.md:76]: Done, added a line here.

File: Getting Started with KMM.md [hunk 6]
```diff
@@ -0,0 +1,564 @@
...
+    UIDevice.currentDevice.systemName() + " " + UIDevice.currentDevice.systemVersion
+```
+
+In the above example, you could notice that we are using platform-specific APIs like `android.os` or
+`UIDevice` in `shared` folder. To keep this organised and readable, KMM has divided the `shared`
+folder into three subfolders - `commonMain`, `androidMain`, `iOSMain`.
+
+![shared folder split](https://mongodb-devhub-cms.s3.us-west-1.amazonaws.com/Screenshot_2023_01_09_at_09_46_57_237bd899da.png)
+
+This is only basic we need before building a complex and full-fledged real app.
```
[the remark this answers is not in the document]
Comment (mohit-sharma-87) [reply Getting Started with KMM.md:120]: Done

File: Getting Started with KMM.md [hunk 7]
```diff
@@ -0,0 +1,564 @@
...
+
+In our viewModel, we have a method `saveQuery` to capture the user queries and share them with the
+speaker. This is information then passed on to our logic layer, `RealmRepo`, which is built using
+Kotlin Multiplatform for Mobile(KMM) as we would like to reuse this for code when building an iOS
+app.
+
+```kotlin
+class RealmRepo {
+
+    suspend fun saveInfo(query: String) {
```
[the remark this answers is not in the document]
Comment (mohit-sharma-87) [reply Getting Started with KMM.md:303]: No, suspend is an integration part of the Kotlin language.

File: Getting Started with KMM.md [hunk 8]
```diff
@@ -0,0 +1,564 @@
...
+With the last step done, we are just one step away from completing the Realm setup. In this step, we
+add the Realm dependency to our project.
+
+Since `Realm` database would be shared across all the platforms, so we would be adding the Realm
+dependency to the common source set of `shared` module. For this we would continue editing our
+`build.gradle` in `shared` module, locate `sourceSet` tag and change accordingly.
+
+From
+
+   ```kotlin
```
[the remark this answers is not in the document]
Comment (mohit-sharma-87) [reply Getting Started with KMM.md:385]: Fixed it

File: Getting Started with KMM.md [hunk 9]
```diff
@@ -0,0 +1,564 @@
...
+integrate this with Atlas Device Sync to save and share our data into the cloud. For doing we
+need follow to 4 simple steps :
+
+1. Create a [free MongoDB account](https://account.mongodb.com/account/login).
+2. Follow the setup wizard after signing up
+   to [create a free cluster](https://www.mongodb.com/docs/atlas/tutorial/deploy-free-tier-cluster/).
+3. Create an
+   [App with App Service UI](https://www.mongodb.com/docs/atlas/app-services/manage-apps/create/create-with-ui/)
+   to enable Atlas Device Sync.
+4. Enable Atlas Device Sync using Flexible Sync, as shown below
```
[the remark this answers is not in the document]
Comment (mohit-sharma-87) [reply Getting Started with KMM.md:469]: Added some additional text..

**Pull request merged**